### PR TITLE
WWST-967 Add DTH for orvibo temp/humidity sensor

### PR DESCRIPTION
**Pull request by FenMei007S**
DTH for orvibo temp/humidity sensor

### CODE REVIEW

File: devicetypes/smartthings/orvibo-temp-humidity-sensor.src/orvibo-temp-humidity-sensor-endpoint.groovy
Comment (ALamchaS) [inline devicetypes/smartthings/orvibo-temp-humidity-sensor.src/orvibo-temp-humidity-sensor-endpoint.groovy]: 1 minute check interval? That's way too short.

File: devicetypes/smartthings/orvibo-temp-humidity-sensor.src/orvibo-temp-humidity-sensor-endpoint.groovy
Comment (ALamchaS) [inline devicetypes/smartthings/orvibo-temp-humidity-sensor.src/orvibo-temp-humidity-sensor-endpoint.groovy]: Why call refresh twice?

**Comment by greens** [comment]
I've been told by one of our hub engineers (@varzac) that we don't really support multi-endpoint zigbee devices. This device might need to be reevaluated.

**Comment by tpmanley** [comment]
I think this device can be made to work since it uses different clusters on the different endpoints. 

I suggest looking into adding support for this device to the `SmartSense Temp/Humidity Sensor`. We try to avoid new DTHs whenever possible and in this case I think that existing DTH can be made to work with a few modifications. Namely `refresh`, `configure` and `ping` will need to be different for the Orvibo device than what is there today. That can be handled by looking at the manufacturer by calling `device.getDataValue("manufacturer")` in those methods.

Another benefit of using this DTH instead of a new one is it runs locally so temperature and humidity events will be generated by the hub and can be used to trigger locally running behaviors.

**Comment by greens** [comment]
@FenMei007S I think you misunderstood Tom. Make changes to the existing file `smartthings/smartsense-temp-humidity-sensor.src/smartsense-temp-humidity-sensor.groovy` by updating its `ping`, `configure`, and `refresh` methods to do different things based on a check of the manufacturer of the device.

**Comment by FenMei007S** [comment]
I checked SmartSense Temp/Humidity Sensor, it looks have different Clusters with orvibo temp/humidity sensor.  Orvibo sensor, Endpoint:
                                             Temperature EP: 1   (0x0000  0x0001 0x0003 0x0009   0x0402)
                                             Humidity EP: 2  (0x0000 0x0003 0x0405)
@greens @tpmanley

**Comment by FenMei007S** [comment]
when i try to modify SmartSense Temp/Humidity Sensor  at IDE ,no any modification be made yet. click Create,error occcurred: Org.springframework.security.access.AccessDeniedException: Run Locally Permission not allowed for DeviceType: f01e3a1b-4d54-4be7-97ff-7a8def12bb2f.

**Comment by FenMei007S** [comment]
For above reason, Is it possabile and necessary to modify existing smartsense-temp-humidity-sensor.groovy to support orvibo device?     By the way, the last commit orvibo-temp-humidity-sensor already work well at test.

**Comment by varzac** [comment]
@FenMei007S for testing yourself, you can change the "runLocally" flag in the header information to false.

**Comment by FenMei007S** [comment]
@varzac. is it possible that sensor with different clusters use the same dth?

**Comment by FenMei007S** [comment]
@greens @tpmanley @varzac  As Tom's suggestion, modifyed existing smartsense-temp-humidity-sensor.groovy  to support orvibo device. As varzac told, "runLocally" be set as false at IDE for self test, it work well. Thanks all of ur suggestion^^, plz review the last code, it need to be meraged to master for further certification test.

File: devicetypes/smartthings/smartsense-temp-humidity-sensor.src/smartsense-temp-humidity-sensor.groovy
Comment (greens) [inline devicetypes/smartthings/smartsense-temp-humidity-sensor.src/smartsense-temp-humidity-sensor.groovy]: add a field here:

`deviceJoinName: "Orvibo Temperature & Humidity Sensor"`

Comment (FenMei007S) [reply devicetypes/smartthings/smartsense-temp-humidity-sensor.src/smartsense-temp-humidity-sensor.groovy]: Updated

**Review by varzac — DISMISSED** [review]

File: devicetypes/smartthings/smartsense-temp-humidity-sensor.src/smartsense-temp-humidity-sensor.groovy
Comment (tpmanley) [inline devicetypes/smartthings/smartsense-temp-humidity-sensor.src/smartsense-temp-humidity-sensor.groovy]: The fingerprint should include the manufacturer as well.

nit: put a space before `deviceId`

File: devicetypes/smartthings/smartsense-temp-humidity-sensor.src/smartsense-temp-humidity-sensor.groovy
Comment (tpmanley) [inline devicetypes/smartthings/smartsense-temp-humidity-sensor.src/smartsense-temp-humidity-sensor.groovy]: coding style nit: there should be spaces after keywords like `if` and `else`

File: devicetypes/smartthings/smartsense-temp-humidity-sensor.src/smartsense-temp-humidity-sensor.groovy
Comment (tpmanley) [inline devicetypes/smartthings/smartsense-temp-humidity-sensor.src/smartsense-temp-humidity-sensor.groovy]: coding style nit: there should be spaces after each `,` separating the parameters

File: devicetypes/smartthings/smartsense-temp-humidity-sensor.src/smartsense-temp-humidity-sensor.groovy
Comment (tpmanley) [inline devicetypes/smartthings/smartsense-temp-humidity-sensor.src/smartsense-temp-humidity-sensor.groovy]: If you think this log message is necessary can you add it to the log at the top of the function (ex: `refresh temperature, humidity and battery for manufacturer Heiman`. Otherwise you can just remove this message altogether.

**Review by tpmanley — COMMENTED** [review]
Just a few minor things but overall is looking good. Can you also squash all the commits and add a Jira to the PR title?

**Comment by FenMei007S** [comment]
@tpmanley @varzac  add manufacturer and update format

**Review by tpmanley — APPROVED** [review]
Changes look good now. Only remaining thing is to squash the commits and the Jira in the PR title.

**Comment by greens** [comment]
I'll do a squash and merge once I get the go-ahead that we're ready to merge these from Magnus. Still trying to figure out our new process with these global cert devices.

**Pull request merged**
